test/testcondition.cpp

```diff
@@ -1851,6 +1851,12 @@ class TestCondition : public TestFixture {
         check("void f1(const std::string &s) { if(s.size() > 42) if(s.empty()) {}} ");
         ASSERT_EQUALS("[test.cpp:1] -> [test.cpp:1]: (warning) Opposite inner 'if' condition leads to a dead code block.\n", errout.str());
 
+        check("void f1(const std::string &s) { if(s.size() > 0) if(s.empty()) {}} ");
+        ASSERT_EQUALS("[test.cpp:1] -> [test.cpp:1]: (warning) Opposite inner 'if' condition leads to a dead code block.\n", errout.str());
+
+        check("void f1(const std::string &s) { if(s.size() < 0) if(s.empty()) {}} ");
```

Comment:
`.size()` cannot <0. So the result seems to be wrong?

Reply:
But you can still write a condition to check for it, which will always be false. I wrote this to get better coverage of the `isZeroBoundCond` function.

Reply:
So
~~~cpp
bool maybe=...
if (false) {
  if (maybe)
    // ...
}
~~~
should issue a `oppositeInnerCondition` warning? That's a FP for me. Maybe warn abut dead code (inside the first if-clause)...

Reply:
But that warning should happen by the library calls. Doesn't cppcheck already have a check for comparing `< 0` against a number that is never negative? Either way its unrelated to the `oppositeInnerCondition`.

Also, it can be the case that size is signed. This happens in libraries because `std::distance` is signed and they don't want to have sign loss. Instead they can do `.size() < 0` to check if there was underflow.